core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -454,15 +454,28 @@ public void writeThrowable(Throwable throwable) throws IOException {
     /**
      * Writes a {@link NamedWriteable} to the current stream, by first writing its name and then the object itself
      */
-    public void writeNamedWriteable(NamedWriteable namedWriteable) throws IOException {
+    public void writeNamedWriteable(@SuppressWarnings("rawtypes") NamedWriteable namedWriteable) throws IOException {
         writeString(namedWriteable.getName());
         namedWriteable.writeTo(this);
     }
 
     /**
+     * Writes an optiona {@link NamedWriteable} to the current stream.
+     * */
+    public void writeOptionalNamedWriteable(@SuppressWarnings("rawtypes") NamedWriteable namedWriteable) throws IOException {
```

Comment:
NamedWritable is a parameterised type so the compiler will complain when used w/o a parameter.

There's a few ways out: Specify the exact type of NamedWriteable that will be written in this location - clearly not an option at compile time here. Specify <?> instead of leaving the raw type - this would trickle down to any classes using this method, including e.g. the list of must clauses in BoolQueryBuilder and imply that all entries in that list need to be of the same type - not a good idea neither. The only way to get rid of the warning I saw was to disable with SuppressWarnings. (javac -X gave the hint on which warnings the JDK I use supports as part of the static analysis)

Checked the codebase for other uses of the annotation found it only in org/elasticsearch/search/aggregations/metrics/ScriptedMetricTests.java and org/elasticsearch/search/aggregations/pipeline/SiblingPipelineAggregator.java so I suppose we don't care too much about it. Happy to remove the annotations I added.